.github/workflows/pull-integration-release.yaml

```diff
@@ -0,0 +1,123 @@
+  pull_request_target:
+    types: [ synchronize, opened, reopened, ready_for_review ]
+    paths-ignore:
+      - "docs/**"
+      - "**/*.md"
+      - "tests/performance/**"
+      - "OWNERS"
+      - "CODEOWNERS"
+      - ".github/**" #Ignoring since build job isn't triggered on the workflow directory changes
+    branches:
+      - 'release-**'
+
+jobs:
+  wait-for-image-build:
+    environment: ${{ github.event.pull_request.author_association != 'COLLABORATOR' && github.event.pull_request.author_association != 'OWNER' && 'restricted' || 'internal' }}
+    name: Wait for image build
+    runs-on: ubuntu-latest
+    if: github.event.pull_request.draft == false
+    steps:
+      - uses: actions/checkout@v4
+        with:
+          fetch-depth: 0
+      - uses: ./.github/actions/wait-for-job-succeed-or-fail
+        with:
+          job-name: 'pull-api-gateway-manager-build'
+          github-auth-token: ${{ secrets.GITHUB_TOKEN }}
+
+  integration-tests:
+    name: Integration tests
+    runs-on: ubuntu-latest
+    if: github.event.pull_request.draft == false
+    needs: [wait-for-image-build]
+    steps:
+      - uses: actions/checkout@v4
+        with:
+          ref: ${{ github.event.pull_request.head.sha }}
+          fetch-depth: 0
+      - uses: ./.github/actions/integration-tests
+        with:
+          client_id: ${{ secrets.CLIENT_ID }}
+          client_secret: ${{ secrets.CLIENT_SECRET }}
+          oidc_issuer_url: ${{ secrets.OIDC_ISSUER_URL }}
+          manager_image: "europe-docker.pkg.dev/kyma-project/dev/api-gateway-manager:PR-${{github.event.number}}"
+
+  k8s-compatibility-check:
+    name: Kubernetes version compatibility test
+    runs-on: ubuntu-latest
+    if: github.event.pull_request.draft == false
+    needs: [ wait-for-image-build ]
+    steps:
+      - uses: actions/checkout@v4
+        with:
+          ref: ${{ github.event.pull_request.head.sha }}
+          fetch-depth: 0
+      - uses: ./.github/actions/integration-tests
+        with:
+          client_id: ${{ secrets.CLIENT_ID }}
+          client_secret: ${{ secrets.CLIENT_SECRET }}
+          oidc_issuer_url: ${{ secrets.OIDC_ISSUER_URL }}
+          manager_image: "europe-docker.pkg.dev/kyma-project/dev/api-gateway-manager:PR-${{github.event.number}}"
+
+  upgrade-tests:
+    name: Upgrade tests
+    runs-on: ubuntu-latest
+    needs: [wait-for-image-build]
+    steps:
+      - uses: actions/checkout@v4
+        with:
+          ref: ${{ github.event.pull_request.head.sha }}
+          fetch-depth: 0
+      - uses: ./.github/actions/upgrade-test
+        with:
+          test_upgrade_img: "europe-docker.pkg.dev/kyma-project/dev/api-gateway-manager:PR-${{github.event.number}}"
+          client_id: ${{ secrets.CLIENT_ID }}
+          client_secret: ${{ secrets.CLIENT_SECRET }}
+          oidc_issuer_url: ${{ secrets.OIDC_ISSUER_URL }}
+          target_branch: ${{ github.base_ref }}
+
+  custom-domain-gcp:
```

Comment:
Custom domain is not supposed to run on PRs

Reply:
It is supposed to run only on PRs to the release branch. This is also why PR provides separate workflow for PRs pointing to the release branches

Reply:
How do we want to support that without `pullRequestTarget`?
Does that mean that we require to do PRs to release branches only from same-repo?

Reply:
OK